### PR TITLE
Remove *model.ChannelList from plugin API return parameter

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
See https://community.mattermost.com/core/pl/euq36dd5ejfbugak97rxcwkxqc

#### Checklist
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by hmhealey** [comment]
I'm all for this since the new types aren't really providing any benefit. I'd really like to change this in the existing code at some point as well, but that would require a major version increase since it would break anyone using client4.go directly

File: app/plugin_api.go [hunk 1]
```diff
@@ -262,8 +262,9 @@ func (api *PluginAPI) GetChannelByNameForTeamName(teamName, channelName string,
 	return api.app.GetChannelByNameForTeamName(channelName, teamName, includeDeleted)
 }
 
-func (api *PluginAPI) GetChannelsForTeamForUser(teamId, userId string, includeDeleted bool) (*model.ChannelList, *model.AppError) {
-	return api.app.GetChannelsForUser(teamId, userId, includeDeleted)
+func (api *PluginAPI) GetChannelsForTeamForUser(teamId, userId string, includeDeleted bool) ([]*model.Channel, *model.AppError) {
+	channels, err := api.app.GetChannelsForUser(teamId, userId, includeDeleted)
+	return *channels, err
```
Comment (lieut-data) [inline app/plugin_api.go]: If `err` is not `nil`, `channels` likely will be and this will panic.

File: app/plugin_api.go [hunk 2]
```diff
@@ -287,8 +291,12 @@ func (api *PluginAPI) UpdateChannel(channel *model.Channel) (*model.Channel, *mo
 	return api.app.UpdateChannel(channel)
 }
 
-func (api *PluginAPI) SearchChannels(teamId string, term string) (*model.ChannelList, *model.AppError) {
-	return api.app.SearchChannels(teamId, term)
+func (api *PluginAPI) SearchChannels(teamId string, term string) ([]*model.Channel, *model.AppError) {
+	channels, err := api.app.SearchChannels(teamId, term)
+	if err != nil {
+		return []*model.Channel{}, err
```
Comment (lieut-data) [inline app/plugin_api.go]: Any reason not to just `return nil, err` here (like on line 269)?

Comment (hanzei) [reply app/plugin_api.go]: No

File: app/plugin_api.go (hunk 1 above)
Comment (hanzei) [reply app/plugin_api.go]: I also missed this https://github.com/mattermost/mattermost-server/pull/9838. Will fix.

**Pull request merged**